csaf-validation/src/main/kotlin/io/github/csaf/sbom/validation/tests/Tests.kt

```diff
@@ -260,6 +269,213 @@ object Test617MultipleScoresWithSameVersionPerProduct : Test {
     }
 }
 
+/**
+ * Implementation of
+ * [Test 6.1.14](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6114-sorted-revision-history).
+ */
+object Test6114SortedRevisionHistory : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        // First, sort items ascending by number
+        val sortedByNumber =
+            doc.document.tracking.revision_history.sortedWith { h1, h2 ->
+                h1.number.compareVersionTo(h2.number)
+            }
+
+        // Then, check if it is sorted by date
+        val isSorted =
+            sortedByNumber.asSequence().zipWithNext { a, b -> a.date <= b.date }.all { it }
+        return if (isSorted) {
+            ValidationSuccessful
+        } else {
+            ValidationFailed(listOf("The revision history is not sorted by ascending date"))
+        }
+        println(isSorted)
+    }
+}
+
+/**
+ * Implementation of
+ * [Test 6.1.16](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6116-latest-document-version).
+ */
+object Test6116LatestDocumentVersion : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        // First, sort items ascending by number
+        val sortedByNumber =
+            doc.document.tracking.revision_history.sortedWith { h1, h2 ->
```

Comment:
Note: the standard would sort by date, not by number. Together with 6.1.14, this should lead to the same result.

Reply:
That's true.. I changed it.

But that brings me to the issue with the test file 6-1-16-03.json
```
"tracking": {
      "current_release_date": "2021-07-21T10:00:00.000Z",
      "id": "OASIS_CSAF_TC-CSAF_2.0-2021-6-1-16-03",
      "initial_release_date": "2021-07-21T10:00:00.000Z",
      "revision_history": [
        {
          "date": "2021-07-21T10:00:00.000Z",
          "number": "2",
          "summary": "Second version."
        },
        {
          "date": "2021-07-21T10:00:00.000Z",
          "number": "1",
          "summary": "Initial version."
        }
      ],
      "status": "final",
      "version": "1"
    }
```

@tschmidtb51 Test 6.1.16 says "It MUST be tested that document version has the same value as the the number in the last item of Revision History when it is sorted ascending by date.". In this case, they have the same date. Is it implied that after the date, they are sorted by number?

Reply:
Yes. Note: Sorting by number takes the versioning scheme into account.

Reply:
Both (6.1.14 and 6.1.16) follow the same logic: date (with timezones) -> number (with versioning scheme)

Reply:
Implemented.